python/pyiceberg/io/fsspec.py

```diff
@@ -112,8 +112,15 @@ def _s3(properties: Properties) -> AbstractFileSystem:
 def _adlfs(properties: Properties) -> AbstractFileSystem:
     from adlfs import AzureBlobFileSystem
 
-    fs = AzureBlobFileSystem(**properties)
-    return fs
+    return AzureBlobFileSystem(
+        connection_string=properties.get("adlfs.connection-string"),
+        account_name=properties.get("adlfs.account-name"),
+        account_key=properties.get("adlfs.account-key"),
+        sas_token=properties.get("adlfs.sas-token"),
+        tenant_id=properties.get("adlfs.tenant-id"),
+        client_id=properties.get("adlfs.client-id"),
+        client_secret=properties.get("adlfs.client-secret"),
+    )
 
 
 SCHEME_TO_FS = {
```

Comment:
I would like to suggest adding here WASB/WASBS support as well:

```
SCHEME_TO_FS = {
    "s3": _s3,
    "s3a": _s3,
    "s3n": _s3,
    "abfs": _adlfs,
    "abfss": _adlfs,
    "wasb": _adlfs,
    "wasbs":_adlfs
}
```

Reply:
I've added them, except the `adl` one. Are you in the position to test the `wasb` endpoint?

Reply:
Hmm, I don't see `wasb` being mentioned in the adlfs repository: https://github.com/search?q=repo%3Afsspec%2Fadlfs%20wasb&type=code

Reply:
I think your changes are not pushed yet.

It looks like they only support:

- az
- abfs
- adl

https://github.com/fsspec/adlfs/blob/2d56e411906519f31ee614e601469f67845b41bd/docs/index.md#fsspec-protocols

I don't know, if they are supporting wasb implicitly. Would need to test. I could take a look tomorrow or on monday.

Reply:
I wasn't sure if `wasb` is supported. I think we should confirm that first

Reply:
Correct.  And the current implementation of Iceberg only supports `abfs` and `az`.  If we want to add `adl`, we would need to add code that checks for protocol and create an instance of `adlfs.AzureBlobFileSystem` or `adlfs.AzureDatalakeFileSystem`.  Right now we only use `adlfs.AzureBlobFileSystem`.

Reply:
I just tested wasb on pandas. Pandas is using adlfs as well. It is not supported:

code:
```
import pandas as pd
storage_options={'account_key': "<account_key>"}

df = pd.read_parquet('wasb://<container_name@<account_name>.blob.core.windows.net/path_to_file.parquet', storage_options=storage_options)
df.head()
```

output:
```
File [c:\Python\Python39\lib\site-packages\pandas\io\parquet.py:493](file:///C:/Python/Python39/lib/site-packages/pandas/io/parquet.py:493), in read_parquet(path, engine, columns, storage_options, use_nullable_dtypes, **kwargs)
    446 """
    447 Load a parquet object from the file path, returning a DataFrame.
    448 
   (...)
    489 DataFrame
    490 """
    491 impl = get_engine(engine)
--> 493 return impl.read(
    494     path,
    495     columns=columns,
    496     storage_options=storage_options,
    497     use_nullable_dtypes=use_nullable_dtypes,
    498     **kwargs,
    499 )

File [c:\Python\Python39\lib\site-packages\pandas\io\parquet.py:233](file:///C:/Python/Python39/lib/site-packages/pandas/io/parquet.py:233), in PyArrowImpl.read(self, path, columns, use_nullable_dtypes, storage_options, **kwargs)
...
--> 208         raise ValueError("Protocol not known: %s" % protocol)
    209     bit = known_implementations[protocol]
    210     try:

ValueError: Protocol not known: wasb
```

The same works with abfs protocol.

Using following url:
`abfs://<container_name@<account_name>.dfs.core.windows.net/path_to_file.parquet`

Reply:
@alaturqua Thanks for giving this a try. I'll leave out wasb for now, and we can pick that up in a separate PR. If you feel that this is something that's pressing, feel free to raise an issue so we can keep track of it.